### DISCUSSION TITLE
Increasing visibility of paternity/maternity leave to support it

### DISCUSSION THREAD

**Discussion by gretacv** [community/community#11089] [Other Feature Feedback, Questions, & Ideas]
Hello,
One of the features I like about github is the widget about contributions. Often I look at it and it gives me a sense of closure and of having achieved a lot. However, when chatting with some developers I learnt that the feeling for those returning from maternity/paternity leave was of disappointment. The sudden emptiness in the squares for many months felt like they hadn't produce anything, when it is quite the opposite. 
I would like to open a discussion on how the homepage of github could be more explicit about supporting those that take a paternity/maternity leave. In the particular case of women in technology it is often the case that one has to choose between forming a family or focusing on ones career. 
I am not sure of what is the best solution, maybe it is to "hide" some months so the break doesn't take too much space, or more celebratory ways like changing the colour of the squares with a different one or even, emojis related to all the work that a baby requires (see image, the idea is not mine). 

![image](https://user-images.githubusercontent.com/11423895/152676108-d678bc3f-6c1a-4786-85e7-5e71d5be0b30.png)

Thank you!

**Comment by raquelalegre** [comment]
Thanks for getting the discussion started, @gretacv! I did get a bit discouraged when I came back from maternity leave and saw my timeline almost completely empty, after having been quite an active contributor for many years. I was lucky enough I had a job to return to, but otherwise this gap could have impacted negatively in a prospective hiring round upon my return to working life. I imagine there are other situations, like returning from long sickness leave, taking a sabbatical, looking after a relative, etc., that could have the same problem. Perhaps another possible solution to add to those suggested by @gretacv is to allow the user to show the latest date they prefer to show on their timeline. At least that'd be a little more discrete and represent better the level of engagement the person has had on GitHub.

PS The plot above was put together by my colleague @tamuri :)

**Reply by gretacv** [reply]
Thanks for sharing your point of view and giving credit where it should be ☺️

**Comment by sadielbartholomew** [comment]
I wanted to comment to say I think this is a really important point @gretacv! I don't have children or caring responsibilities in my life at the moment, but if/when I do this would be a big concern too. And whilst I in no way mean to distract from the specific issue at hand, namely:

> a discussion on how the homepage of github could be more explicit about supporting those that take a paternity/maternity leave. In the particular case of women in technology it is often the case that one has to choose between forming a family or focusing on ones career.

I do think that the best solution to the main issue you are discussing, namely the resultant gaps in the contributions graph, could perhaps best be covered by a more general solution, one that, as @raquel-ucl points out, might cover such situations that could also leave big gaps:

> I imagine there are other situations, like returning from long sickness leave, taking a sabbatical, looking after a relative, etc., that could have the same problem.

I prefer the idea of some means to annotate the contributions graph and in particular gaps from lack of contributions, rather than hiding parts of it, but maybe the option to do either is ideal (or maybe my suggestion below about marking given days as unavailable covers both bases, in a way).

> or more celebratory ways like changing the colour of the squares with a different one or even, emojis related to all the work that a baby requires (see image, the idea is not mine).

I like these, though personally I think the best solution would be something a bit more subtle and indicative of days where development isn't possible, like the ability to 'white out' (or equivalently 'black out' if using a dark background) the squares on days which are effectively unavailable for any development work that would otherwise lead to contributions, with an optional hover-over note to annotate as to why those days were marked as development no-go days. I think that would help too towards the wider issue of the unhealthy nature of the contributions graph which implicitly gives the message that every day is free for development and the pressure that we should try to have at least one contribution every day!

(Of course, days aren't the granularity of events in real life, so it's not going to be perfect, but that's how the contributions graph works already so we can't change that part...)

Another idea that I don't think has been mentioned is that, given contributions are very flexible in nature, it is possible to use scripts or cron jobs to create 'dummy' or trivial contributions, so instead of ending up with a large gap when you know you will have a long-term contribution drought, you could create and use some sort of script to make a daily commit to a local repo (maybe even to update your own GitHub user profile with a running tally of days spent in maternity or paternity, say) so that you don't end up with a contributions gap in the first place.

So perhaps someone could create a basic library that allows anyone to set that up very easily (to configure such a script as appropriate) so they can create the contributions that represent the amount of development (in contribution terms) they would otherwise have made, ideally in a meaningful way (e.g. milestones for baby age maternity or paternity, or recovery from illness, if the person was happy to share). But strictly this is avoiding the issue of gaps in the first place!

Those are my initial thoughts on ideas for the contributions graph, but overall I am looking forward to following this thread and hearing more about the main point:

> I would like to open a discussion on how the homepage of github could be more explicit about supporting those that take a paternity/maternity leave.

**Reply by gretacv** [reply]
Thank you so much for sharing your ideas! A package that somehow is like having an "out of the office" message could be useful. Not sure how over engineered that might be, but I am not an expert.

**Reply by rkillick** [reply]
I really like the "out of office" idea.  When I went on maternity leave I was still getting github questions and comments on my repos but it looked like I wasn't a responsive person.  If people emailed me they would have seen my out of office but instead they didn't get anything.  An OoO feature would be useful as most people who just didn't have time to contribute probably wouldn't put an out of office on, but those who do, could then have a different coloured square in the contributions graph.  You could even go one step further and have those who do commit when an out of office is on in a different colour too - to encourage people to not respond when they have given a signal they can't.  This could also be used for annual leave and normalize that people do and should be encouraged to take leave.  The barrier to putting on an out of office would also be lower and less subject to abuse than an automated bot making commits.  If it becomes "normal" to see the colour on the graph too it would make those who use it for longer periods of time more likely to use it.

**Reply by alanocallaghan** [reply]
Massive +1 for the out of office idea. I feel bad when returning from annual leave as I often miss issues opened over that time. Bad for maintainers who may miss bugs, PRs, etc, and bad for contributors as they feel they're being ignored. This must be orders of magnitude worse for longer absences like parental leave.

**Reply by eduardobonet** [reply]
> like the ability to 'white out' (or equivalently 'black out' if using a dark background) the squares on days which are effectively unavailable for any development work that would otherwise lead to contributions, with an optional hover-over note to annotate as to why those days were marked as development no-go days

I really like the idea of allowing users to display why they couldn't contribute do so, as a way to humanize this whole process. When I see a blank contribution space marked as OoO, sure that's ok, but if I see they mark they went on vacation or a (m|p)aternity leave marker, suddenly you get reminded that those are people that have their lives, and that's heartwarming. It's just not saying "I wasn't available". It's "I wasn't available because I had this amazing things happening in my life".

**Reply by tupui** [reply]
The white-out idea is great! I did a world travel and just could not do anything for some month. It would have been great to indicate that it should not be expected to see any contributions for a period of time.

Also I would like GH to acknowledge	other tasks on this graph. Things like comment, triaging, etc. Sometimes it takes even more time and energy than making commits and really not revealing of the effort we put.

**Reply by gretacv** [reply]
I have just discovered "status" by clicking on my profile picture. Maybe this feature could be extended so whatever the status it is reflected into the "contribution" view?
![image](https://user-images.githubusercontent.com/11423895/154864348-f5b44e3e-73d4-46ec-bc7c-e913d333a450.png)

**Comment by eduardobonet** [comment]
@gretacv This is such a great discussion, took the liberty to bring into GitLab as well: https://gitlab.com/gitlab-org/gitlab/-/issues/352487
